### ISSUE TITLE
Build failure: syslinux-unstable and xen and libvirt after staging-next 2024-12-04 #361878

### ISSUE THREAD

**Issue by spiage** [NixOS/nixpkgs#367703]
## Steps To Reproduce

1. build libvirt

## Build log

```
note: keeping build directory '/tmp/nix-build-syslinux-unstable-2019-02-07.drv-2/build'
error: builder for '/nix/store/wxw7sqwsskv6kg7b6j20zai8fhzkmxh9-syslinux-unstable-2019-02-07.drv' failed with exit code 2;
       last 25 log lines:
       > /build/source/com32/chain/chain.c:172:35: warning: cast from pointer to integer of different size []
       >   172 |                                   (addr_t) data[i].data, data[i].size))
       >       |                                   ^
       > /build/source/com32/chain/chain.c:246:53: warning: cast from pointer to integer of different size []
       >   246 |         if (syslinux_add_movelist(&mlist, endimage, (addr_t) swapstub,
       >       |                                                     ^
       > /build/source/com32/chain/chain.c: In function 'main':
       > /build/source/com32/chain/chain.c:517:44: error: passing argument 3 of 'loadfile' from incompatible pointer type []
       >   517 |         if (loadfile(opt.file, &fdat.data, &fdat.size)) {
       >       |                                            ^~~~~~~~~~
       >       |                                            |
       >       |                                            addr_t * {aka unsigned int *}
       > In file included from /build/source/com32/chain/chain.c:32:
       > /build/source/com32/include/syslinux/loadfile.h:11:37: note: expected 'size_t *' {aka 'long unsigned int *'} but argument is of type 'addr_t *' {aka 'unsigned int *'}
       >    11 | int loadfile(const char *, void **, size_t *);
       >       |                                     ^~~~~~~~
       > make[4]: *** [/build/source/com32/chain/Makefile:28: chain.o] Error 1
       > make[4]: Leaving directory '/build/source/efi64/com32/chain'
       > make[3]: *** [/build/source/com32/Makefile:8: chain] Error 2
       > make[3]: Leaving directory '/build/source/efi64/com32'
       > make[2]: *** [/build/source/Makefile:288: com32] Error 2
       > make[2]: Leaving directory '/build/source/efi64'
       > make[1]: *** [/build/source/Makefile:272: efi64] Error 2
       > make[1]: Leaving directory '/build/source'
       > make: *** [Makefile:102: bios] Error 2
       For full logs, run 'nix log /nix/store/wxw7sqwsskv6kg7b6j20zai8fhzkmxh9-syslinux-unstable-2019-02-07.drv'.
```

## Notify maintainers

@fpletz 
@globin 
@lovesegfault

---

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by spiage** [comment]
@fpletz @globin @lovesegfault
Can you make zen as an option, please?

**Comment by spiage** [comment]
@gador Thank you!
Fixed with #367948
